### PR TITLE
Measles template

### PR DESCRIPTION
**Pull request by kdoyle514**
Updated cholera template for measles usage.

### CODE REVIEW

File: inst/rmarkdown/templates/measles_outbreak/skeleton/skeleton.Rmd [hunk 1]
```diff
@@ -464,10 +464,8 @@ linelist_cleaned$age_group_mon <- age_categories(linelist_cleaned$age_months,
                                                  ceiling = TRUE)
 
 ## create an age group variable by specifying categorical breaks
-## categories are 0-5m, 6-11m, 12-23m, 24-59m, 5-14y, 15-29y, 30-44y, 45+
 linelist_cleaned$age_group <- age_categories(linelist_cleaned$age_years, 
-                                             breakers = c(0, 0.5, 1, 2, 5,
-                                                          15, 30, 45))
+                                             breakers = c(0, 5, 15, 30, 45))
```
Comment (aspina7) [inline inst/rmarkdown/templates/measles_outbreak/skeleton/skeleton.Rmd]: I changed the 5 yr age group to 3, this makes the lowest age category 0-2yrs which allows us to merge with the month variable above.

Comment (aspina7) [reply inst/rmarkdown/templates/measles_outbreak/skeleton/skeleton.Rmd]: also added breaks to conform with isidros requests of age-groups

File: inst/rmarkdown/templates/measles_outbreak/skeleton/skeleton.Rmd [hunk 2]
```diff
@@ -881,10 +889,9 @@ mortality_rate(deaths, population, multiplier = 10000, mergeCI = TRUE) %>%
   kable(digits = 2)
 ```
 
-To produce a crude mortality rate per 10,000 people per day, use the folowing 
+To produce a crude mortality rate attributable to measles per 10,000 people per day, use the folowing 
 code chunk. 
-This assumes that you are capturing every death in your population and that 
-your population remains stable over the time period of interest. 
+This assumes that you are capturing every measles death in your population and that your population remains stable over the time period of interest.
```
Comment (aspina7) [inline inst/rmarkdown/templates/measles_outbreak/skeleton/skeleton.Rmd]: These linebreaks were actually in there on purpose  (makes your rmd easier to read and dont show up as line breaks in your word doc). 
To show a line indicating 80 characters - go to globaloptions > code > displace and click show margin

File: inst/rmarkdown/templates/measles_outbreak/skeleton/skeleton.Rmd [hunk 3]
```diff
@@ -1116,27 +1120,30 @@ cowplot::plot_grid(
 ```
 
 
-We could then also look at admissions and exits by epiweek for operational purposes. 
+We could then also look at admissions and exits among inpatients by epiweek for operational purposes.
```
Comment (aspina7) [inline inst/rmarkdown/templates/measles_outbreak/skeleton/skeleton.Rmd]: actually what we should do here is group by inpatient/outpatient and show admissions that way. Then show exits by inpatients only.

**Comment by aspina7** [comment]
Looks good so far @kdoyle514 
Been through it and theres one discussion point on the admissions per week stuff, see [here](https://github.com/R4EPI/sitrep/pull/97/files#r266237216)

Also there appears to be an error when plotting the age pyramid for under 2s. Dont have time to figure it out right now though. maybe @zkamvar ?

**Comment by aspina7** [comment]
@zkamvar two things remaining open here for measles. Fixing age_pyramid and fixing group_age_categories. 

* [ ] group_age_categories: if age_months NA and age_years == 0, then you get a 0-2 years category. What you actually want is the lowest level of age_months (I would say....) - not strictly correct, but guess thats what need to go on... 
* [ ] age_pyramid, i would say we just drop NAs in the grouping var too so there is no bar plotted but the age level is still shown (empty). See the error in age_group_mon bar plot where sex levels not enough...

**Comment by zkamvar** [comment]
> group_age_categories: if age_months NA and age_years == 0, then you get a 0-2 years category. What you actually want is the lowest level of age_months (I would say....) - not strictly correct, but guess thats what need to go on...

I'm not sure what would create this situation. Why would you have age_months as NA and yet age_years as 0 without age_days being filled in?

**Comment by aspina7** [comment]
> > group_age_categories: if age_months NA and age_years == 0, then you get a 0-2 years category. What you actually want is the lowest level of age_months (I would say....) - not strictly correct, but guess thats what need to go on...
> 
> I'm not sure what would create this situation. Why would you have age_months as NA and yet age_years as 0 without age_days being filled in?

because i only used age_years and age_months in group_age_categories - and i didnt fill forward age_days in to age_months (age_months not filled in where age_days is) - have done that now so it works...

**Comment by aspina7** [comment]
But actually even if you fill forward - the 0-2 years level shows up but has zero count...

**Comment by zkamvar** [comment]
Will I be able to reproduce this behavior from the template?

**Comment by aspina7** [comment]
Yeah from measles if you dont run the factor relevel line of code after running group-age-categories

**Comment by aspina7** [comment]
merging measles for now - the fixes above are still needed but want to start a meningitis template so kate can keep going. Then can go back and change cholera/measles once those fixes are implemented.

**Pull request merged**